elements/rh-tile/rh-tile.ts

```diff
@@ -141,6 +141,7 @@ export class RhTile extends LitElement {
   override async willUpdate(changed: PropertyValues<this>) {
     this.#internals.role = this.radioGroup ? 'radio' : this.checkable ? 'checkbox' : null;
     this.#internals.ariaChecked = !this.#isCheckable ? null : String(!!this.checked);
+    this.#internals.ariaLabel = !this.#isCheckable ? null : 'headline';
```

Comment:
see test output. this line causes the tile to announce as 'headline', whereas previously it announced as 'Tile 1'. I believe the test case below is the correct behaviour, meaning that the element's text content should be the label. I will revert this line but preserve the rest of the fixes from this commit, and please review: am i seeing this clearly?

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/1466420/add5c943-e0aa-4ef6-a05d-36c2ecacb1e4)

Reply:
@nikkimk see https://github.com/RedHat-UX/red-hat-design-system/pull/1302/commits/6e68ff0de350510a25b8b3f7cdc0f15f234854eb

Reply:
@bennypowers according to our Guidelines in docs, the heading slot is what should be used: https://ux.redhat.com/elements/tile/guidelines/#heading

Reply:
Ok, we can adjust the code which computes the label to only consider elements slotted into the headline slot. 

but what if there's no headline text, we still need the accessible-label escape hatch, right?

Reply:
Let's confirm with @marionnegp, but if I'm not mistaken selectable tiles should always have a headline. 

https://xd.adobe.com/view/925ed24c-3ddf-4111-bacf-2a42e5249f7a-c70d/screen/16207897-40cc-4e49-b064-2940e769b11b

Reply:
Yes, the headline should be the text for the option you're selecting.

Reply:
It's it possible to have a selectable tile without a headline?

Reply:
Selectable tiles aren't supposed to have images or icons, so they should always have a text headline.